### PR TITLE
Write and test docstrings for models

### LINKED ISSUE #110: ENH: does get_L need is_sparse ?
It can be inferred from X, as X is passed to get_L

### PR DESCRIPTION
**Pull request by mathurinm**
After writing this I think Lasso, ElasticNet, WeightedLasso and Logistic regression could inherit from a common class. This would make a lot of code less duplicated.

I think it allowed me to catch a few bugs. I have written lots of TODO.

closes #110

### CODE REVIEW

File: sparse_ho/models/enet.py [hunk 1]
```diff
@@ -236,24 +263,47 @@ def _get_jac_t_v(X, y, jac, mask, dense, alphas, v, n_samples):
...
+        """Compute Lipschitz constant of datafit.
+
+        Parameters
+        ----------
+        X: np.array-like, shape (n_samples, n_features)
+            Design matrix.
+
+        Returns
+        -------
+        L: float
```
Comment (mathurinm) [inline sparse_ho/models/enet.py:302]: in fact those are coordinatewise lipschitz const, right ?

File: sparse_ho/models/enet.py [hunk 2]
```diff
@@ -392,12 +392,12 @@ def get_jac_v(self, X, y, mask, dense, jac, v):
...
-    def get_hessian(X_train, y_train, mask, dense, log_alpha):
+    def get_hessian(X, y_train, mask, dense, log_alpha):
         """Compute Hessian of datafit.
 
         Parameters
         ----------
-        X_train: np.array-like, shape (n_samples, n_features)
+        X: np.array-like, shape (n_samples, n_features)
             Design matrix.
         y_train: np.array, shape (n_samples,)
```
Comment (mathurinm) [inline sparse_ho/models/enet.py:398]: @QB3 also

Comment (QB3) [reply sparse_ho/models/enet.py]: done

**Comment by mathurinm** [comment]
@Klopfe as per @QB3 the remaining todos concern functions written mostly for SVM and SVR that you know better, so we leave it for another PR when you have time

**Pull request merged**